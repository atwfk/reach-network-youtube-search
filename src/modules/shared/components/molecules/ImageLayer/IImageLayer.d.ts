declare namespace IImageLayer {
  export interface IProps {
    src: string;
    alt: string;
    width?: string;
    height?: string;
    duration?: string;
    playlistCount?: number;
  }
}

export { IImageLayer };
