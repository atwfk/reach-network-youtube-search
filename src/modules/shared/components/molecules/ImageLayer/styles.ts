import styled from "styled-components";

export const StyledImageLayer = styled.div`
  position: relative;
  color: #f1f0f0;

  .duration {
    position: absolute;
    bottom: 10px;
    right: 10px;
    background-color: rgba(0, 0, 0, 0.7);
    padding: 1px;
  }

  .playlist-count {
    position: absolute;
    top: 1px;
    right: 0;
    width: 40%;
    height: 97%;
    background-color: rgba(0, 0, 0, 0.7);
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;

    .playlist-icon {
      font-size: 25px;
      margin-top: 6px;
    }
  }
`;
