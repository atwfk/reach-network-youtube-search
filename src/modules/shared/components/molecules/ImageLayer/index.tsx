import React from "react";
import type { FC, ReactElement } from "react";
import Image from "../../atoms/Image";
import { IImageLayer } from "./IImageLayer";
import { CgPlayList } from "react-icons/cg";
import { StyledImageLayer } from "./styles";

const ImageLayer: FC<IImageLayer.IProps> = ({
  src,
  alt,
  width,
  height,
  duration,
  playlistCount
}): ReactElement => {
  return (
    <StyledImageLayer className="image-container">
      <Image alt={alt} height={`${height}`} width={`${width}`} src={src} />
      {duration && (
        <span className="duration" data-testid="duration">
          {duration}
        </span>
      )}
      {playlistCount && (
        <span className="playlist-count" data-testid="playlist-count">
          {playlistCount}
          <CgPlayList className="playlist-icon" />
        </span>
      )}
    </StyledImageLayer>
  );
};

export default ImageLayer;
