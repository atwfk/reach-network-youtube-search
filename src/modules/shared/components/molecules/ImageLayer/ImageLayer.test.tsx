import { render, screen, cleanup } from "@testing-library/react";
import "jest-styled-components";
import renderer from "react-test-renderer";
import ImageLayer from ".";

afterEach(cleanup);

describe("ImageLayer Component snapshots", () => {
  it("Should render ImageLayer component with duration when we pass duration prop", () => {
    const tree = renderer
      .create(
        <ImageLayer
          src={"https://i.ytimg.com/vi/X48VuDVv0do/mqdefault.jpg"}
          alt="Kubernetes Tutorial for Beginners [FULL COURSE in 4 Hours]"
          width="320"
          height="180"
          duration="03:36:55"
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("Should render ImageLayer component with number of playlist videos when we pass playlistCount prop", () => {
    const tree = renderer
      .create(
        <ImageLayer
          src={"https://i.ytimg.com/vi/X48VuDVv0do/mqdefault.jpg"}
          alt="Kubernetes Tutorial for Beginners [FULL COURSE in 4 Hours]"
          width="320"
          height="180"
          playlistCount={13}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe("Stats Component RTL", () => {
  it("Should have duration when we pass duration prop", () => {
    render(
      <ImageLayer
        src={"https://i.ytimg.com/vi/X48VuDVv0do/mqdefault.jpg"}
        alt="Kubernetes Tutorial for Beginners [FULL COURSE in 4 Hours]"
        width="320"
        height="180"
        duration="03:36:55"
      />
    );

    const duration = screen.getByTestId("duration");
    const playlistCount = screen.queryByTestId("playlist-count");

    expect(duration).toBeInTheDocument();
    expect(playlistCount).not.toBeInTheDocument();
  });

  it("Should have playlist count when we pass playlistCount prop", () => {
    render(
      <ImageLayer
        src={"https://i.ytimg.com/vi/X48VuDVv0do/mqdefault.jpg"}
        alt="Kubernetes Tutorial for Beginners [FULL COURSE in 4 Hours]"
        width="320"
        height="180"
        playlistCount={13}
      />
    );

    const playlistCount = screen.getByTestId("playlist-count");
    const duration = screen.queryByTestId("duration");

    expect(playlistCount).toBeInTheDocument();
    expect(duration).not.toBeInTheDocument();
  });
});
