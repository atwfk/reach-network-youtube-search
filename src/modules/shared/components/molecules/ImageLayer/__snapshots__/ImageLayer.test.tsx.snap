// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageLayer Component snapshots Should render ImageLayer component with duration when we pass duration prop 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 .channel {
  border-radius: 50%;
}

.c0 {
  position: relative;
  color: #f1f0f0;
}

.c0 .duration {
  position: absolute;
  bottom: 10px;
  right: 10px;
  background-color: rgba(0,0,0,0.7);
  padding: 1px;
}

.c0 .playlist-count {
  position: absolute;
  top: 1px;
  right: 0;
  width: 40%;
  height: 97%;
  background-color: rgba(0,0,0,0.7);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .playlist-count .playlist-icon {
  font-size: 25px;
  margin-top: 6px;
}

<div
  className="c0 image-container"
>
  <div
    className="c1"
  >
    <img
      alt="Kubernetes Tutorial for Beginners [FULL COURSE in 4 Hours]"
      className=""
      data-testid="image"
      height="180"
      src="https://i.ytimg.com/vi/X48VuDVv0do/mqdefault.jpg"
      width="320"
    />
  </div>
  <span
    className="duration"
    data-testid="duration"
  >
    03:36:55
  </span>
</div>
`;

exports[`ImageLayer Component snapshots Should render ImageLayer component with number of playlist videos when we pass playlistCount prop 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 .channel {
  border-radius: 50%;
}

.c0 {
  position: relative;
  color: #f1f0f0;
}

.c0 .duration {
  position: absolute;
  bottom: 10px;
  right: 10px;
  background-color: rgba(0,0,0,0.7);
  padding: 1px;
}

.c0 .playlist-count {
  position: absolute;
  top: 1px;
  right: 0;
  width: 40%;
  height: 97%;
  background-color: rgba(0,0,0,0.7);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .playlist-count .playlist-icon {
  font-size: 25px;
  margin-top: 6px;
}

<div
  className="c0 image-container"
>
  <div
    className="c1"
  >
    <img
      alt="Kubernetes Tutorial for Beginners [FULL COURSE in 4 Hours]"
      className=""
      data-testid="image"
      height="180"
      src="https://i.ytimg.com/vi/X48VuDVv0do/mqdefault.jpg"
      width="320"
    />
  </div>
  <span
    className="playlist-count"
    data-testid="playlist-count"
  >
    13
    <svg
      className="playlist-icon"
      fill="none"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M16 5H4V7H16V5Z"
        fill="currentColor"
      />
      <path
        d="M16 9H4V11H16V9Z"
        fill="currentColor"
      />
      <path
        d="M4 13H12V15H4V13Z"
        fill="currentColor"
      />
      <path
        d="M20 16L14 13V19L20 16Z"
        fill="currentColor"
      />
    </svg>
  </span>
</div>
`;
